packages/nimble-components/src/theme-provider/index.ts

```diff
@@ -33,19 +54,44 @@ export const theme = DesignToken.create<Theme>({
  * @internal
  */
 export class ThemeProvider extends FoundationElement {
-    @attr({
-        attribute: 'direction'
-    })
+    @attr()
+    public override lang = '';
+
+    @attr()
     public direction: Direction = Direction.ltr;
 
-    @attr({
-        attribute: 'theme'
-    })
+    @attr()
     public theme: Theme = Theme.light;
 
+    public get validity(): ValidityObject {
+        return {
+            invalidLang: this.langIsInvalid
+        };
+    }
+
+    private langIsInvalid = false;
+
+    public langChanged(
+        _prev: string | undefined | null,
+        next: string | undefined | null
+    ): void {
+        this.langIsInvalid = false;
+        if (
+            next !== undefined
+            && next !== null
+            && next !== ''
```

Comment:
It looks like you're checking for all falsey values here, so why not just write this as
```suggestion
            next
```